src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -1383,4 +1402,18 @@ public UUID getUuid() {
 		return getUUID();
 	}
 
+	public long getJoinedNationAt() {
+		return joinedNationAt;
+	}
+
+	public void setJoinedNationAt(long joinedNationAt) {
+		this.joinedNationAt = joinedNationAt;
+	}
+
+	private void sortResidents() {
+		List<Resident> sortedResidents = new ArrayList<Resident>(residents.stream().sorted(Comparator.comparingLong(Resident::getJoinedTownAt)).collect(Collectors.toList()));
```

Comment:
Don't need the `new ArrayList<Resident>()` part, the stream will collect it to a list already.